internal/rbd/rbd_util.go

```diff
@@ -532,23 +532,26 @@ func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 func (rv *rbdVolume) getCloneDepth(ctx context.Context) (uint, error) {
 	var depth uint
 	vol := rbdVolume{}
-	defer vol.Destroy()
 
 	vol.Pool = rv.Pool
 	vol.Monitors = rv.Monitors
 	vol.RbdImageName = rv.RbdImageName
 	vol.conn = rv.conn.Copy()
 
-	err := vol.openIoctx()
-	if err != nil {
-		return depth, err
-	}
-
 	for {
 		if vol.RbdImageName == "" {
 			return depth, nil
 		}
+		err := vol.openIoctx()
+		if err != nil {
+			return depth, err
+		}
+
 		err = vol.getImageInfo()
+		// FIXME: create and destroy the vol inside the loop.
+		// see https://github.com/ceph/ceph-csi/pull/1838#discussion_r598530807
```

Comment:
This really is dangerous. Maybe file a cleanup PR to do this and rebase this PR on top of that.

Or, create an issue for this problem as this really needs correcting urgently.

Reply:
Created an issue for the same https://github.com/ceph/ceph-csi/issues/2144 . Since this issue exists in the current implementation and not something that this PR introduces, I don't think we need to block this PR for that.

Reply:
yeah, imo, it can be a followup PR.